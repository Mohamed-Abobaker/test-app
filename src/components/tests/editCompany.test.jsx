import React from 'react';
import { shallow } from 'enzyme';
import EditCompany from '../editCompany';

describe('EditCompany.jsx tests', () => {
  it('should render the correct component', () => {
    const wrapper = shallow(<EditCompany />)

    expect(wrapper.debug()).toMatchSnapshot();
  });
  it('should call correct function when form is submitted', () => {
    const mockFunc = jest.fn();
    const mockFunc1 = jest.fn();
    const updateFromMock = jest.fn();

    const wrapper = shallow(<EditCompany updateCompanies={mockFunc} company={{ companyName: '', sector: '', creationDay: '', country: '', customers: 0, active: false }} index={0} companies={[{ companyName: '', sector: '', creationDay: '', country: '', customers: 0, active: false }, {}, {}]} updateCompanyIndex={mockFunc1} />)
    const forms = wrapper.find('form');
    forms.at(0).props().onSubmit()

    expect(mockFunc).toHaveBeenCalled()
    expect(mockFunc1).toHaveBeenCalled()
  });
});