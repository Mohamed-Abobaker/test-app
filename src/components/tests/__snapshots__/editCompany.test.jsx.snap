// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditCompany.jsx tests should render the correct component 1`] = `
"<div>
  <form onSubmit={[Function: onSubmit]}>
    <label>
      Name
      :
      <input required={true} type=\\"text\\" value={[undefined]} onChange={[Function: onChange]} />
    </label>
    <label>
      Country
      :
      <input required={true} type=\\"text\\" value={[undefined]} onChange={[Function: onChange]} />
    </label>
    <label>
      Creation date
      :
      <input required={true} type=\\"date\\" value={[undefined]} onChange={[Function: onChange]} />
    </label>
    <label>
      sector
      :
      <input required={true} type=\\"text\\" value={[undefined]} onChange={[Function: onChange]} />
    </label>
    <label>
      Number of customers:
      <input required={true} type=\\"number\\" min={0} value={[undefined]} onChange={[Function: onChange]} />
    </label>
    <label>
      Active:
      <select value={[undefined]} onChange={[Function: onChange]}>
        <option value={true}>
          true
        </option>
        <option value={false}>
          false
        </option>
      </select>
    </label>
    <div>
      <button type=\\"submit\\">
        Save
      </button>
    </div>
  </form>
</div>"
`;
